File: Algorithms/p274_H-Index_1.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        citations.sort(reverse=True)
        cnt = 0
        for v in citations:
            cnt += 1
            if v<cnt:
                cnt -= 1
            if v<=cnt:
                break
        return cnt

File: Algorithms/p274_H-Index/p274_H-Index_1.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        sort(citations.begin(),citations.end(),greater<int>());
        int cnt = 0;
        for(int v : citations){
            ++cnt;
            if(v<cnt) --cnt;
            if(v<=cnt) break;
        }
        return cnt;
    }
};
